src/swap.rs

```diff
@@ -0,0 +1,140 @@
+use std::convert::TryInto;
+
+use lightning::ln::PaymentHash;
+use rgbstd::contract::ContractId;
+
+use crate::utils::hex_str_to_vec;
+
+#[derive(Debug, Clone, Copy)]
+pub enum SwapType {
+    BuyAsset { amount_rgb: u64, amount_msats: u64 },
+    SellAsset { amount_rgb: u64, amount_msats: u64 },
+}
+
+impl SwapType {
+    pub fn opposite(self) -> Self {
+        match self {
+            SwapType::BuyAsset {
+                amount_rgb,
+                amount_msats,
+            } => SwapType::SellAsset {
+                amount_rgb,
+                amount_msats,
+            },
+            SwapType::SellAsset {
+                amount_rgb,
+                amount_msats,
+            } => SwapType::BuyAsset {
+                amount_rgb,
+                amount_msats,
+            },
+        }
+    }
+
+    pub fn is_buy(&self) -> bool {
+        matches!(self, SwapType::BuyAsset { .. })
+    }
+
+    pub fn amount_msats(&self) -> u64 {
+        match self {
+            SwapType::BuyAsset { amount_msats, .. } | SwapType::SellAsset { amount_msats, .. } => {
+                *amount_msats
+            }
+        }
+    }
+    pub fn amount_rgb(&self) -> u64 {
+        match self {
+            SwapType::BuyAsset { amount_rgb, .. } | SwapType::SellAsset { amount_rgb, .. } => {
+                *amount_rgb
+            }
+        }
+    }
+
+    pub fn side(&self) -> &'static str {
+        match self {
+            SwapType::BuyAsset { .. } => "buy",
+            SwapType::SellAsset { .. } => "sell",
+        }
+    }
```

Comment:
please remove this method and instead implement the `Display` trait for the `SwapType` enum